3rdparty/crypto/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ add_library(crypto sha256_avx2.cpp
                    sha256_sse4.cpp
                    sha256_sse41.cpp
                    sha256.cpp
```

Comment:
I don't think `CSHA256` is used anywhere in the codebase after this PR. We should remove the SHA256 source files.